client/src/Components/Filter/Filter.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react'
+import { useNavigate } from 'react-router-dom'
+
+import './Filter.css'
+
+const Filter = () => {
+    const [formState, setFormState] = useState({
+        maxPrice: '',
+        includes: '',
+        limit: ''
+    })
+    const navigate = useNavigate()
+
+    const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setFormState(prev => ({ ...prev, [name]: value }))
+    };
+
+    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        const queryParams = new URLSearchParams(formState).toString();
+        navigate(`/products?${queryParams}`)
+    };
+
+    return (
+        <form className='filter' onSubmit={handleSubmit}>
+            <label>
+                Max Price: 
+                <input
+                    type='text'
```

Comment:
I would change this to a number input. It prevents users from trying to search with words. We always want to prevent users from doing anything crazy. You can still treat a number input as text on this end until it absolutely needs to become a number.